### PR TITLE
feat: Add Logging API entry points and central handler

### PR DESCRIPTION
**Pull request by metal-messiah**
Add Logging API interface and central handling mechanism as customer interface for capturing logging data
---

### Related Issue(s)
https://new-relic.atlassian.net/browse/NR-251068
https://new-relic.atlassian.net/browse/NR-251074

### Testing
Jest unit tests have been updated to check for EE events.  WDIO tests to be added after feature/havesting is implemented

### CODE REVIEW

File: src/loaders/api/api.js [hunk 1]
```diff
@@ -51,6 +51,23 @@ export function setAPI (agentIdentifier, forceDrain, runSoftNavOverSpa = false)
...
+   * @param {string} level
+   */
+  function log (message, context, level = 'info') {
+    handle(SUPPORTABILITY_METRIC_CHANNEL, [`API/log${level}/called`], undefined, FEATURE_NAMES.metrics, instanceEE)
+    handle('log', [now(), message, context, level], undefined, FEATURE_NAMES.logging, instanceEE)
+  }
+
+  logApiMethods.forEach((method) => {
+    apiInterface[method] = function (message, context) {
+      log(message, context, method.toLowerCase().replace('log', ''))
```
Comment (patrickhousley) [inline src/loaders/api/api.js:67]:
```suggestion
    apiInterface[method] = function (message, additionalProps) {
      log(message, additionalProps, method.toLowerCase().replace('log', ''))
```

File: src/loaders/api/api.js [hunk 2]
```diff
@@ -51,6 +51,23 @@ export function setAPI (agentIdentifier, forceDrain, runSoftNavOverSpa = false)
...
 
+  /**
+   *
+   * @param {string} message
+   * @param {{[key: string]: *}} context
+   * @param {string} level
+   */
+  function log (message, context, level = 'info') {
+    handle(SUPPORTABILITY_METRIC_CHANNEL, [`API/log${level}/called`], undefined, FEATURE_NAMES.metrics, instanceEE)
+    handle('log', [now(), message, context, level], undefined, FEATURE_NAMES.logging, instanceEE)
```
Comment (patrickhousley) [inline src/loaders/api/api.js:62]:
```suggestion
  function log (message, additionalProps, level = 'info') {
    handle(SUPPORTABILITY_METRIC_CHANNEL, [`API/log${level}/called`], undefined, FEATURE_NAMES.metrics, instanceEE)
    handle('log', [now(), message, additionalProps, level], undefined, FEATURE_NAMES.logging, instanceEE)
```

**Review by patrickhousley — APPROVED** [review]
Just a couple comments on our favorite word `context`

File: src/loaders/api/api.js (hunk 2 above)
Comment (metal-messiah) [reply src/loaders/api/api.js:62]: These props are actually changed in the sister PR, will leave alone for now to avoid merge conflict

**Pull request merged**